tensorboard/plugins/profile/tf_profile_dashboard/tf-profile-dashboard.html

```diff
@@ -222,19 +240,26 @@ <h3>No profile data was found.</h3>
       if (oldDatasets != null || this.selected == null) {
         // Select the first dataset by default.
         this.set('selectedDataset', 0);
+        this.set('currentSelected', 0);
       }
     },
     _selectedDatasetChanged: function(newDataset, oldDataset) {
       if (this._datasets) {
         // Display the first tool by default when switching to another run.
-        this.set('selectedTool', 0);
+        this.set('currentSelected', (newDataset << 16));
+
+        // Same tool can have differernt index in different runs, therefore we force a change of
+        // 'selectedTool', to make sure everything is synced.
+        this.async(function() {
+          this.set('selectedTool', undefined);
+          this.set('selectedTool', 0);
+        }.bind(this));
       }
     },
-    _getCurrentToolName: function(selectedToolIndex, activeTools) {
-      if (selectedToolIndex >= 0 && selectedToolIndex < activeTools.length) {
-        return activeTools[selectedToolIndex];
+    _selectedToolChanged: function(newTool, oldTool) {
+      if (this._datasets && !(newTool === undefined)) {
+        this.set('currentSelected', ((this.selectedDataset << 16) | newTool));
```

Comment:
Could you add a comment noting what this bit shift is doing?

Reply:
done